packages/python/plotly/plotly/express/_core.py

```diff
@@ -374,17 +374,27 @@ def configure_cartesian_marginal_axes(args, fig, orders):
         set_cartesian_axis_opts(args, xaxis, "x", orders)
 
     # Configure axis ticks on marginal subplots
+    # Don't change the marginal grid settings if the template already specifies `showgrid`
+    if isinstance(
+        pio.templates[args["template"]].layout.xaxis.showgrid, bool
```

Comment:
here you should check `fig.layout.template` actually

Reply:
Sorry, that's not right... you can just check `args["template"]` directly, at this point it should already be a template object.

Reply:
Sorry, that's not right either! but it should be... I'm going to PR that in.

Reply:
I removed the `pio.templates` part to be compatible with your upcoming PR. I also changed the comparison to be against `None` for clarity.